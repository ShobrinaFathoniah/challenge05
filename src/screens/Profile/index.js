import {StyleSheet, Image, View, TouchableOpacity} from 'react-native';
import React from 'react';
import {useDispatch, useSelector} from 'react-redux';
import {moderateScale} from 'react-native-size-matters';
import {LibreBaskerville} from '../../components/Fonts';
import {Header} from '../../components';
import {BLACK, DARK_PURPLE_300, RED_500, WHITE} from '../../helpers/colors';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import AntDesign from 'react-native-vector-icons/AntDesign';
import {
  sendDataLoginWithGoogle,
  setDataUser,
  setDataUserWithGoogle,
} from '../Login/redux/action';
import {loginPic} from '../../assets';
import {useIsFocused} from '@react-navigation/native';
import analytics from '@react-native-firebase/analytics';

const Profile = ({navigation}) => {
  const {dataUserGoogle, dataUser} = useSelector(state => state.login);
  const dispatch = useDispatch();
  const isFocused = useIsFocused();

  const focused = async () => {
    if (isFocused) {
      console.log('focused');

      await analytics().logScreenView({
        screen_class: 'Profile',
        screen_name: 'Profile',
      });
    } else {
      console.log('not focused');
    }
  };

  focused();

  const signInWithGoogle = () => {
    dispatch(sendDataLoginWithGoogle(navigation));
    dispatch(setDataUser({user: null}));
  };

  const googleButton = () => {
    return (
      <TouchableOpacity
        style={[styles.button, styles.containerButtonLogout]}
        onPress={signInWithGoogle}>
        <FontAwesome name="google" color={RED_500} size={24} />
      </TouchableOpacity>
    );
  };

  const logOut = () => {
    dispatch(setDataUser({user: null}));
    dispatch(setDataUserWithGoogle({user: null}));
    navigation.navigate('Login');
  };

  const logOutButton = () => {
    return (
      <TouchableOpacity
        style={[styles.button, styles.containerButtonLogout]}
        onPress={logOut}>
        <AntDesign name="logout" color={DARK_PURPLE_300} size={24} />
      </TouchableOpacity>
    );
  };

  const data = () => {
    if (dataUserGoogle.user !== null) {
      return (
        <View>
          <Image
            source={{uri: dataUserGoogle.user.photo}}
            style={styles.image}
          />
          <View style={styles.containerName}>
            <LibreBaskerville style={styles.textNama}>
              {dataUserGoogle.user.name}
            </LibreBaskerville>
          </View>
          <View style={styles.containerName}>
            <LibreBaskerville style={styles.textNama}>
              {dataUserGoogle.user.email}
            </LibreBaskerville>
          </View>
          {logOutButton()}
        </View>
      );
    } else if (dataUser.user !== null) {
      return (
        <View>
          <Image
            source={{
              uri: dataUser.photoURL
                ? dataUser.photoURL
                : 'https://www.jobstreet.co.id/en/cms/employer/wp-content/plugins/all-in-one-seo-pack/images/default-user-image.png',
            }}
            style={styles.image}
          />
          <View style={styles.containerName}>
            <LibreBaskerville style={styles.textNama}>
              {dataUser.displayName ? dataUser.displayName : 'Nama Anda'}
            </LibreBaskerville>
          </View>
          <View style={styles.containerName}>
            <LibreBaskerville style={styles.textNama}>
              {dataUser.phoneNumber}
            </LibreBaskerville>
          </View>
          <View style={styles.containerName}>
            <LibreBaskerville style={styles.textNama}>
              {dataUser.email ? dataUser.email : googleButton()}
            </LibreBaskerville>
          </View>
        </View>
      );
    } else {
      return (
        <View style={styles.containerNotLogin}>
          <Image source={loginPic} style={styles.imagePic} />
          <LibreBaskerville style={styles.text}>
            You're login using fingerprint, to see this page, you must login
            using Google Account
          </LibreBaskerville>
          <View style={styles.twoColumns}>
            {googleButton()}
            {logOutButton()}
          </View>
        </View>
      );
    }
  };

  return (
    <View style={styles.container}>
      <Header text="Profile" radiusBottom={true} />
      {data()}
    </View>
  );
};

export default Profile;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  image: {
    width: moderateScale(150),
    height: moderateScale(150),
    alignSelf: 'center',
    borderRadius: moderateScale(100),
    marginVertical: moderateScale(10),
  },
  containerName: {
    margin: moderateScale(10),
    alignSelf: 'center',
  },
  textNama: {
    fontSize: moderateScale(18),
    color: BLACK,
  },
  button: {
    padding: moderateScale(10),
    borderRadius: moderateScale(3),
    backgroundColor: WHITE,
    margin: moderateScale(10),
  },
  containerButtonLogout: {
    alignSelf: 'center',
  },
  text: {
    alignSelf: 'center',
    fontSize: moderateScale(14),
    margin: moderateScale(10),
    color: BLACK,
    textAlign: 'center',
  },
  containerNotLogin: {
    flex: 1,
  },
  imagePic: {
    height: moderateScale(300),
    width: moderateScale(300),
    alignSelf: 'center',
    margin: moderateScale(10),
  },
  twoColumns: {
    flexDirection: 'row',
    alignSelf: 'center',
  },
});
