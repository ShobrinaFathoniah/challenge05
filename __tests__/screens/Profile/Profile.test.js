jest.useFakeTimers();

import React from 'react';
import Profile from '../../../src/screens/Profile';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../../src/helpers/reduxTesting';

describe('Test Snapshoot', () => {
  test('Profile Snapshoot', () => {
    jest.useFakeTimers();

    const snap = create(ContainerTesting(<Profile />));
    expect(snap).toMatchSnapshot();
  });
});
