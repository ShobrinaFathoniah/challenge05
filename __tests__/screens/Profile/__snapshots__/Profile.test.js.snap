// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshoot Profile Snapshoot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#8180b7",
        "borderBottomEndRadius": 150,
        "borderBottomStartRadius": 150,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "PassionsConflict-Regular",
          },
          Object {
            "color": "#000000",
            "fontSize": 66,
            "letterSpacing": 3.142857142857143,
            "margin": 15.714285714285714,
            "textAlign": "center",
          },
        ]
      }
    >
      Profile
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/Image/creativity.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 471.42857142857144,
          "margin": 15.714285714285714,
          "width": 471.42857142857144,
        }
      }
    />
    <Text
      ellipsizeMode="tail"
      numberOfLines={0}
      style={
        Array [
          Object {
            "fontFamily": "LibreBaskerville-Regular",
          },
          Object {
            "alignSelf": "center",
            "color": "#000000",
            "fontSize": 22,
            "margin": 15.714285714285714,
            "textAlign": "center",
          },
        ]
      }
    >
      You're login using fingerprint, to see this page, you must login using Google Account
    </Text>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4.714285714285714,
            "margin": 15.714285714285714,
            "opacity": 1,
            "padding": 15.714285714285714,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#c4131e",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4.714285714285714,
            "margin": 15.714285714285714,
            "opacity": 1,
            "padding": 15.714285714285714,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#535487",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
